File: MetodyObliczeniowe/MetodyObliczeniowe/Wezel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MetodyObliczeniowe
{
	public class Wezel : ICloneable
	{
		#region public Properties

		public int Potega { get; set; }
		public string Wspolczynnik { get; set; } //An
		public int WspolczynnikPomocniczy = 1;
		public string X { get; set; }
		public double WartoscWezla { get; set; } //Trzyma obliczona juz wartosc wezla

		#endregion

		#region Method

		public Wezel(int potega, string x = "x")
		{
			X = x;
			Potega = potega;
			Wspolczynnik = "a" + Potega;
		}

		public override string ToString()
		{
			if (WspolczynnikPomocniczy == 0 || Wspolczynnik.Equals("0"))
			{
				return "";
			}
			double wspolczynnik = 0.0;
			if (!Wspolczynnik.Contains("a"))
				Double.TryParse(Wspolczynnik, out wspolczynnik);

			string rownianie = string.Format("{0}", Wspolczynnik.Contains("a") ? Wspolczynnik : Math.Round(wspolczynnik, 2).ToString());

			if (WspolczynnikPomocniczy != 1)
				rownianie += string.Format("*{0}", WspolczynnikPomocniczy);

			rownianie += string.Format("*{0}^{1}", X, Potega);

			return rownianie;
		}

		public object Clone()
		{
			return this.MemberwiseClone();
		}

		#endregion Method
	}
}

File: zadanie_1/lib/Wezel.rb
class Wezel
  attr_accessor :Potega
  attr_accessor :Wspolczynnik
  attr_accessor :WspolczynnikPomocniczy
  attr_accessor :X
  attr_accessor :WartoscWezla

  def initialize(potega, x = "x", wspolczynnik = "", wspPom = 1.0) 
    if(IsValidParameters(potega, x, wspolczynnik, wspPom))
      @Potega = potega
      @X = x
      @WspolczynnikPomocniczy = wspPom
      @WartoscWezla = 0
      if(wspolczynnik == "")
        @Wspolczynnik = "a#{@Potega}"
      else
        @Wspolczynnik = wspolczynnik
      end
    else
      raise "Invalid argument"
    end
  end

  def ToString 
    if (@WspolczynnikPomocniczy == 0.0 || @Wspolczynnik == "0.0")
      return
    end

    if(@Wspolczynnik.to_s.include? "a")
      rownianie = "#{@Wspolczynnik}"
    else
      wynik = @Wspolczynnik.to_f
      rownianie = "#{wynik.round(2)}"
    end

    if(@WspolczynnikPomocniczy != 1)
      rownianie += "*#{@WspolczynnikPomocniczy}"
    end

    rownianie += "*#{@X}^#{@Potega}"
  end

  private
  
    def IsValidParameters(potega, x, wspolczynnik, wspPom)
      isValid = true
    
      if(!potega.is_a? Fixnum)
        isValid = false
      end

      if(!x.is_a? String)
        isValid = false
      end
    
      if(!wspolczynnik.is_a? String)
        isValid = false
      end

      if(!wspPom.is_a? Float)
        isValid = false
      end

      return isValid
    end
end